geniza/annotations/signals.py

```diff
@@ -0,0 +1,34 @@
+from geniza.annotations.models import Annotation
+from geniza.corpus.models import Document
+
+
+def update_footnote(instance, **kwargs):
+    """On Annotation save or delete, update related footnote's has_transcription field
+    and trigger a reindex of related documents (via footnote save)"""
+    created = kwargs.get("created")
+    deleted = created is None
+    # get canvas URI from annotation content and match it to Documents
+    canvas_uri = instance.content["target"]["source"]["id"]
+    docs = Document.objects.filter(fragments__manifest__canvases__uri=canvas_uri)
+    # get source URI from annotation content and match it to document's Footnotes
+    source_uri = instance.content["dc:source"]
+    source_pk = int(source_uri.split("/")[-1])
+    # get other annotations on this canvas and source
+    other_annotations_exist = (
+        Annotation.objects.exclude(pk=instance.pk)
+        .filter(
+            content__target__source__id=canvas_uri,
+            content__contains={"dc:source": source_uri},
+        )
+        .count()
+        > 0
+    )
```

Comment:
Actually, since you're connecting to the `post_delete` signal, the deleted annotation isn't in the database so this definitely doesn't need the exclusion and we can just check with `.exists()`

We should only do this query if the signal is a delete.

Reply:
Sounds like we should remove this altogether if we're not going to do anything on deleting the last annotation, but I'll save this here in case we decide we need it:

```python
    # get other annotations on this manifest and source
    other_annotations_exist = (
        Annotation.objects.filter(
            content__target__source__partOf__id=manifest,
            content__contains={"dc:source": source_uri},
        )
        .exists()
    )
```